includes/assets/js/NFDPluginsMarketplace.js

```diff
@@ -1,99 +1,128 @@
+          } else {
+            this.isLoaded();
+            this.showProducts(this.filterProducts(data.products.data));
+          }
+        });
+    });
+  }
+
+  // Function that sets up the container for the plugin marketplace
+  setupContainer() {
+    const pluginsListTable = document.createElement('div');
+    pluginsListTable.classList.add(
+      'wp-list-table',
+      'widefat',
+      'plugin-install',
+      'nfd-premium-plugins-marketplace'
+    );
+    pluginsListTable.innerHTML = `<h2 class="screen-reader-text">Plugins list</h2>
         <p>${nfdPremiumPluginsMarketplace.marketplaceDescription}</p>
         <div id="the-list">${this.isLoading()}</div>`;
 
-        const filterArea = document.querySelector('.plugin-install-tab-premium-marketplace .wp-filter');
-        filterArea.parentNode.insertBefore(pluginsListTable, filterArea.nextSibling);
-    }
-
-    // Filter products to only show plugins
-    filterProducts(products) {
-        const plugins = products.filter(product => product.type == "plugin");
-
-        return plugins;
+    const filterArea = document.querySelector(
+      '.plugin-install-tab-premium-marketplace .wp-filter'
+    );
+    filterArea.parentNode.insertBefore(
+      pluginsListTable,
+      filterArea.nextSibling
+    );
+  }
+
+  // Filter products to only show plugins
+  filterProducts(products) {
+    const plugins = products.filter((product) => product.type == 'plugin');
+
+    return plugins;
+  }
+
+  // Returns whether the environment supports CTB
+  supportsCTB(products) {
+    if (ctbAvailableProducts.includes(products?.clickToBuyId)) return true;
+    if (typeof nfdctb === 'undefined') {
+      return false;
     }
 
-     // Returns whether the environment supports CTB
-    supportsCTB() {
-        if (typeof nfdctb === 'undefined') {
-            return false;
-        }
-
-        return nfdctb.supportsCTB;
-    }
-
-    // Show the filtered products on the page
-    showProducts(products) {
-        
-        // Helper function to render the thumbnail for each product
-        const renderThumbnail = (product) => {
-            if (product.smallThumbnailUrl) {
-                return `<img src="${product.smallThumbnailUrl}" alt="${product.name} Thumbnail" width="128px" height="128px">`;
-            }
-
-            const fallback = 'https://hiive-space-cdn.nyc3.cdn.digitaloceanspaces.com/marketplace/premium-plugins/fallback.svg';
-            return `<img src="${fallback}" alt="${product.name} Thumbnail" width="128px" height="128px">`;
-        }
-
-        // Helper function to render the action buttons for each product
-        const renderActions = (product) => {
-            const supportsCTB = this.supportsCTB();
-            let actionButtons = [];
-
-            if (product.primaryUrl && product.price) {
-                const primaryAction = `<a target="_blank" 
+    return nfdctb.supportsCTB;
+  }
+
+  // Show the filtered products on the page
+  showProducts(products) {
```

Comment:
This `supportsCTB` shouldn't be needed. We can safely add the CTB attributes and the ctb module will detect whether it is supported or not.

FYI, there is a legacy ctb module and a newer global ctb module. They both will detect the same attributes and which one (if any) is supported.